File: changeIp.js
const {execSync} = require('child_process');
const settings = require('electron-settings');

function changeToSchool() {
    return new Promise(async resolve => {
        const adp = await settings.get('adp');
        const ip = await settings.get('ip');
        const mask = "255.255.255.0";
        const gate = await settings.get('gate');
        const dns1 = '211.46.153.1';
        const dns2 = '210.104.203.1';
        try {
            execSync('netsh -c int ip set address name="' + adp + '" source=static addr=' + ip + ' mask=' + mask + ' gateway=' + gate + ' gwmetric=0');
            execSync('netsh -c int ip set dnsservers name="' + adp + '" source=static ' + dns1 + ' validate=no');
            execSync('netsh interface ip add dns name="' + adp + '" addr=' + dns2 + ' validate=no index=2');
        } catch (e) {

        }
        await settings.set('stat', 1);
        resolve();
    });
}

function changeToOut() {
    return new Promise(async resolve => {
        const adp = await settings.get('adp');
        try {
            execSync('netsh -c int ip set address name="' + adp + '" source=dhcp');
            execSync('netsh -c int ip set dnsservers name="' + adp + '" source=dhcp');
        } catch (e) {

        }
        await settings.set('stat', 0);
        resolve();
    });
}

async function getCurrentState() {
    const settings = require('electron-settings');
    const t = await settings.get('stat');
    if (!t) return 0;
    else return t;
}

exports.changeToSchool = changeToSchool;
exports.changeToOut = changeToOut;
exports.getCurrentState = getCurrentState;

File: src/backend/changeIp.py
import os


def runSafe(command):
    try:
        os.system(command)
    except:
        pass


def changeIp(place, adp, stuid):
    if place == 1:  # home
        runSafe('netsh -c int ip set address name="' + adp + '" source=dhcp')
        runSafe('netsh -c int ip set dnsservers name="' + adp + '" source=dhcp')
    elif place == 2:  # school
        if len(stuid) == 4:
            a = int(stuid[0])
            b = int(stuid[1])
            c = int(stuid[2:])
        else:
            a = int(stuid[0])
            b = int(stuid[2])
            c = int(stuid[3:])

        su = 9 + c + (b - 1) * 16
        if a == 1:
            ip = '10.140.82.' + str(su)
            gate = '10.140.82.254'

        elif a == 2:
            ip = '10.140.83.' + str(su)
            gate = '10.140.83.254'

        else:
            ip = '10.140.84.' + str(su)
            gate = '10.140.84.254'

        mask = "255.255.255.0"
        dns1 = '211.46.153.1'
        dns2 = '210.104.203.1'
        runSafe(
            'netsh -c int ip set address name="' + adp + '" source=static addr=' + ip + ' mask=' + mask + ' gateway='
            + gate + ' gwmetric=0')
        runSafe('netsh -c int ip set dnsservers name="' + adp + '" source=static ' + dns1 + ' validate=no')
        runSafe('netsh interface ip add dns name="' + adp + '" addr=' + dns2 + ' validate=no index=2')
